packages/node/src/indexer/dictionary/v2/ethDictionaryV2.spec.ts

```diff
@@ -0,0 +1,254 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import {
+  NodeConfig,
+  BlockHeightMap,
+  IBlock,
+  DictionaryResponse,
+} from '@subql/node-core';
+import {
+  EthereumBlock,
+  EthereumDatasourceKind,
+  EthereumHandlerKind,
+  SubqlRuntimeDatasource,
+} from '@subql/types-ethereum';
+import {
+  EthereumProjectDs,
+  SubqueryProject,
+} from '../../../configure/SubqueryProject';
+import {
+  buildDictionaryV2QueryEntry,
+  EthDictionaryV2,
+} from './ethDictionaryV2';
+
+const HTTP_ENDPOINT = 'https://polygon.api.onfinality.io/public';
+const mockDs: EthereumProjectDs[] = [
+  {
+    kind: EthereumDatasourceKind.Runtime,
+    assets: new Map(),
+    startBlock: 3678215,
+    mapping: {
+      file: './dist/index.js',
+      handlers: [
+        {
+          handler: 'handleTransaction',
+          kind: EthereumHandlerKind.Call,
+          filter: {
+            function: 'approve(address spender, uint256 rawAmount)',
+          },
+        },
+        {
+          handler: 'handleLog',
+          kind: EthereumHandlerKind.Event,
+          filter: {
+            topics: [
+              'Transfer(address indexed from, address indexed to, uint256 amount)',
+            ],
+          },
+        },
+      ],
+    },
+  },
+];
+
+const nodeConfig = new NodeConfig({
+  subquery: 'polygon-starter',
+  subqueryName: 'polygon-starter',
+  dictionaryTimeout: 10,
+  networkEndpoint: [HTTP_ENDPOINT],
+  networkDictionary: ['http://localhost:3000/rpc'],
+});
+
+const m = new Map<number, any>();
+mockDs.forEach((ds, index, dataSources) => {
+  m.set(ds.startBlock, dataSources.slice(0, index + 1));
+});
+const dsMap = new BlockHeightMap(m);
+
+// enable this once dictionary v2 is online
+describe('eth dictionary v2', () => {
+  let ethDictionaryV2: EthDictionaryV2;
+  let ethBlock3678215: EthereumBlock;
+  let ethBlock3678250: EthereumBlock;
+  beforeEach(async () => {
+    ethDictionaryV2 = await EthDictionaryV2.create(
+      'http://localhost:3000/rpc',
```

Comment:
We should only define the url in one place so we can easily switch from a takoyaki endpoint to a geth one